src/implementations/svd.jl

```diff
@@ -206,10 +206,37 @@ function svd_vals!(A::AbstractMatrix, S, alg::LAPACK_SVDAlgorithm)
     return S
 end
 
-function svd_trunc!(A, USVᴴ, alg::TruncatedAlgorithm)
+# nothing case here to handle GenericLinearAlgebra
+function svd_trunc!(A, USVᴴϵ::Tuple{TU, TS, TVᴴ, Tϵ}, alg::TruncatedAlgorithm) where {TU, TS, TVᴴ, Tϵ}
+    U, S, Vᴴ, ϵ = USVᴴϵ
+    U, S, Vᴴ = svd_compact!(A, (U, S, Vᴴ), alg.alg)
+    USVᴴtrunc, ind = truncate(svd_trunc!, (U, S, Vᴴ), alg.trunc)
+    if !isempty(ϵ)
+        ϵ[1] = truncation_error!(diagview(S), ind)
+    end
+    return USVᴴtrunc..., ϵ
+end
+function svd_trunc!(A, USVᴴϵ::Tuple{Nothing, Tϵ}, alg::TruncatedAlgorithm) where {Tϵ}
+    USVᴴ, ϵ = USVᴴϵ
     U, S, Vᴴ = svd_compact!(A, USVᴴ, alg.alg)
     USVᴴtrunc, ind = truncate(svd_trunc!, (U, S, Vᴴ), alg.trunc)
-    return USVᴴtrunc..., truncation_error!(diagview(S), ind)
+    if !isempty(ϵ)
+        ϵ[1] = truncation_error!(diagview(S), ind)
+    end
+    return USVᴴtrunc..., ϵ
+end
+
+function svd_trunc!(A, USVᴴ::Tuple{TU, TS, TVᴴ}, alg::TruncatedAlgorithm; compute_error::Bool = true) where {TU, TS, TVᴴ}
+    Tr = real(eltype(A))
+    ϵ = compute_error ? zeros(Tr, 1) : zeros(Tr, 0)
```

Comment:
Wouldn't it be easier and more GPU friendly to have 
```suggestion
    ϵ = similar(S, compute_error)
```